### PR TITLE
feat: switch to 0.60.0-rhobs1 PO fork 

### PR DESCRIPTION
**Pull request by sthaha**
This PR switches from using platform PO to using the forked PO tailored for Observability Operator. 
Switching from 0.57.0 to 0.60.0 introduces the following changes ...

* switch to go 1.18 (and associated changes to build)
* Changes to dependencies
* Changes to CRD (api-groups)

Switching to RHOBS fork only requires changes to api-group i.e. monitoring.coreos -> monitoring.rhobs.

**NOTE:** This fork does not ship `admission-webhook` and will be done as followup commit. 

Signed-off-by: Sunil Thaha [sthaha@redhat.com](mailto:sthaha@redhat.com)

### CODE REVIEW

File: pkg/reconciler/reconciler.go
Comment (simonpasquier) [inline pkg/reconciler/reconciler.go:33]: Why log the error and return it to the caller? In general I'm a big proponent of "Only handle errors once" (https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully).

Comment (sthaha) [reply pkg/reconciler/reconciler.go:33]: The issue is that when error gets logged, we have no idea `which` reconciler had issues reconciling `what`.

e.g. see if you can find the issue here and reason out why: https://github.com/rhobs/observability-operator/actions/runs/3231587984/jobs/5291269766

Comment (sthaha) [reply pkg/reconciler/reconciler.go:33]: Case in point, ... 
https://github.com/rhobs/observability-operator/actions/runs/3232742256/jobs/5293721088#step:4:971

```
2022-10-12T07:23:49Z	INFO	thanos-querier	watched MonitoringStack changed, checking for matching querier	{"Monitoring Stack": "e2e-tests/valid-loglevel"}
2022-10-12T07:23:49Z	INFO	observability-operator	patch failed	{"stack": "e2e-tests/valid-loglevel", "reconciler": "updater", "name": "valid-loglevel-alertmanager", "ns": "e2e-tests", "error": "roles.rbac.authorization.k8s.io \"valid-loglevel-alertmanager\" not found"}
2022-10-12T07:23:49Z	INFO	observability-operator	Failed to get prometheus object	{"stack": "e2e-tests/valid-loglevel", "err": "Prometheus.monitoring.rhobs \"valid-loglevel\" not found"}
2022-10-12T07:23:49Z	ERROR	Reconciler error	{"controller": "monitoringstack", "controllerGroup": "monitoring.rhobs", "controllerKind": "MonitoringStack", "monitoringStack": {"name":"valid-loglevel","namespace":"e2e-tests"}, "namespace": "e2e-tests", "name": "valid-loglevel", "reconcileID": "a60ef5f6-b16e-4dc4-82b5-91ad97f4f3f4", "error": "roles.rbac.authorization.k8s.io \"valid-loglevel-alertmanager\" not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234
2022-10-12T07:23:49Z	INFO	observability-operator	Reconciling monitoring stack	{"stack": "e2e-tests/valid-loglevel"}
```

* The CI run failed with the following error - NOTE: `reconciler` is `updater`  and place it failed was when calling `Patch` which is a surprise (to me).

Comment (simonpasquier) [reply pkg/reconciler/reconciler.go:33]: My idea is that the error message should be as descriptive as possible. For this particular example:

```
return fmt.Errorf("%s/%s (%s): failed to set owner reference: %w", r.resource.GetNamespace(), r.resource.GetName(), r.resource.GetObjectKind().GroupVersionKind().String(), err)
```

Comment (sthaha) [reply pkg/reconciler/reconciler.go]: Yes, that will work and I will revert adding logger to including these details in the error msg. Thanks Simon 🤗

**Review by jan--f — APPROVED** [review]
Nice, lgtm!

**Comment by sthaha** [comment]
Thanks @jan--f 🤗 
I will wait for @simonpasquier to  approve as well before merging and creating a release

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**
